### DISCUSSION TITLE
Frigate+ docker run command using PLUS_API_KEY

### DISCUSSION THREAD

**Discussion by lukasig** [blakeblackshear/frigate#8790] [General]
I install frigate using this docker run command:

```
sudo docker run --privileged -d \
  --name frigate \
  --restart=unless-stopped \
  --mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 \
  --device /dev/bus/usb:/dev/bus/usb \
  --device /dev/dri/renderD128 \
  --shm-size=256m \
  --network host \
  -v /media/usb/frigate/media:/media/frigate \
  -v /home/username/frigate/config/config.yml:/config/config.yml \
  -v /etc/localtime:/etc/localtime:ro \
  -e FRIGATE_RTSP_PASSWORD='redacted' \
  -p 5000:5000 \
  -p 8554:8554 \
  -p 8555:8555/tcp \
  -p 8555:8555/udp \
  ghcr.io/blakeblackshear/frigate:stable
```
I am trying to add my Frigate+ api key to the docker run command
`  -e PLUS_API_KEY: '9XXXXXXa-1234-4XXXXXXXe-7XXXXXXXXXX9: ......restofAPIkey......'`

but  I keep getting the error:

> docker: Error response from daemon: pull access denied for 9XXXXXXa-1234-4XXXXXXXe-7XXXXXXXXXX9, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.

I noticed that the API keys have a ":" somwhere in the middle and that is what the docker error might have a problem with.

What am I missing?

frigate version 0.12.1-367d724
using docker under ubuntu server 22.04

thanks

**Comment by NickM-27** [comment]
it seems you most likely added the -e after the image, the image must be the last thing in the list

**Reply by lukasig** [reply]
that is not the case
this is what the full command looks like:
```
sudo docker run --privileged -d \
  --name frigate \
  --restart=unless-stopped \
  --mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 \
  --device /dev/bus/usb:/dev/bus/usb \
  --device /dev/dri/renderD128 \
  --shm-size=256m \
  --network host \
  -v /media/usb/frigate/media:/media/frigate \
  -v /home/username/frigate/config/config.yml:/config/config.yml \
  -v /etc/localtime:/etc/localtime:ro \
  -e FRIGATE_RTSP_PASSWORD='redacted' \
  -p 5000:5000 \
  -p 8554:8554 \
  -p 8555:8555/tcp \
  -p 8555:8555/udp \
  -e PLUS_API_KEY: '9XXXXXXa-1234-4XXXXXXXe-7XXXXXXXXXX9: ......restofAPIkey......'
  ghcr.io/blakeblackshear/frigate:stable
```

**Reply by NickM-27** [reply]
the syntax is incorrect, it should be `-e PLUS_API_KEY='123xyz'` just like the environment variable above

**Reply by lukasig** [reply]
that is also not the issue. Both variants should work but none do.

BTW...could this be related to frigate version 0.12?

**Reply by NickM-27** [reply]
the way I described works just fine for me

> BTW...could this be related to frigate version 0.12?

no, frigate is not started yet so this has nothing to do with frigate

**Reply by lukasig** [reply]
have tried again using your suggestion and now i get this error:

> "docker run" requires at least 1 argument.
> See 'docker run --help'.
> 
> Usage:  docker run [OPTIONS] IMAGE [COMMAND] [ARG...]
> 
> Create and run a new container from an image
> -bash: ghcr.io/blakeblackshear/frigate:stable: No such file or directory

**Reply by NickM-27** [reply]
you'll need to include the command you're running

**Reply by lukasig** [reply]
```
sudo docker run --privileged -d \
  --name frigate \
  --restart=unless-stopped \
  --mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 \
  --device /dev/bus/usb:/dev/bus/usb \
  --device /dev/dri/renderD128 \
  --shm-size=256m \
  --network host \
  -v /media/usb/frigate/media:/media/frigate \
  -v /home/username/frigate/config/config.yml:/config/config.yml \
  -v /etc/localtime:/etc/localtime:ro \
  -p 5000:5000 \
  -p 8554:8554 \
  -p 8555:8555/tcp \
  -p 8555:8555/udp \
  -e PLUS_API_KEY='xxx'
  ghcr.io/blakeblackshear/frigate:stable
```

**Reply by NickM-27** [reply]
you are missing the `\` after the PLUS_API_KEY='xxx'

**Reply by lukasig** [reply]
gee

kind of funny

thanks man
